Self Assessment: start / end screens, buttons, pickers, headers, backgrounds

jeffstolz opened the pull request:
#### Description:

Draft progress PR that styles the self-assessment updated designs displayed here https://www.figma.com/file/79OwaKqHvSeriExpYzAvVN/PATHS-Exploration-April-and-May-2020?node-id=3489%3A23320

Includes:
- Start / end screen designs
- Buttons
- Pickers
- Headers
- Typography
- Page backgrounds
- Bypass captcha screens
- Refactoring existing code
- Stripping out old illustrations

#### Screenshots:

![self-assessment](https://user-images.githubusercontent.com/2924479/83812415-7f9dcd80-a689-11ea-8f2a-a8f942e1f37c.gif)

johnschoeman committed:
Update ios productname/headers to separate targets

Why:
We would like to be able to build the BT version of the app without
including any GPS location specific code. Currently there is some
location specific code in AppDelegate.m and some location specific swift
files being included in the BT build. We would like to start the process
of segmenting the application code by target.

This commit:
Moves Location specific code to the GPS group. It separates
AppDelegate.m into two AppDelegates, one for each target. We set up
swift bridging headers for each target. Additionally we restored some
defaults to the library and framework search paths and updated the
Product names to match the Target names of GPS and BTE as this is the
default that is expected by xcode build process and allows us to use the
default search paths provided by xcode.

Note that this commit changes no functionality.

Note that the BTE target is still building with GPS only pods currently
that are being defined from the react-native side and will need to be
removed in the BTE target in a future commit. Specifically RN
LocationAlways and @maurn85/react-native-background-geolocation

tstirrat committed:
chore(CI) Append git SHA to CI/PR build version (#873)

MichaelNeas committed:
SCrypt performance time and staging build settings (#939)

tstirrat committed:
Revert "chore(CI) Append git SHA to CI/PR build version (#873)" (#940)

This reverts commit 9605c94c54f676f31f2b7df8d241d3d78df577ef.

JacobJaffe committed:
E2E Transfer Flow (#934)

* export through verification code

* error handling

* e2e functionality on apiary endpoints. feature flag the old and new screens

* en for export start

* Select HA localization

* localize code input and consent screens

* Localize confirm and complete

* Accesibility between screen sizes

* exit warning alert

* update spec route

* updated to use themes where possible, and PR cleanup

* add feature flag to staging

* remove array spread

johnschoeman committed:
Update detox scripts to use correct product

Why:
A previous commit updated the product names for the various builds and
the detox test scripts were not updated to the new names.

This commit:
Updates the detox scripts to the correct names.

johnschoeman committed:
Use xcode 11.5 for workflow ios builds

Why:
We would like to be able to use the swift package manager and be on the
latest version of xcode to have the most up to date support and
security patches. Currently all of our github actions are building with
xcode 11.

This commit:
Updates the various actions to use Xcode_11.5.app, which is the most
recent version available on github virtual environments at the moment:

https://github.com/actions/virtual-environments/blob/43e26fa96b8e36c96dc4f825b349bd417625b92c/images/macos/macos-10.15-Readme.md

Note that for us to be able to use the ExposureNotification entitlement
recently released by apple we will need to build with Xcode 11.5 beta,
which isn't available on github actions at the moment.

Tambet-Ingo-RocketInsights committed:
Time Android location hashing (#936)

Adding signpost logs to measure the performance of geohash+scrypt hashing in android

johnschoeman committed:
Introduce typescript

Why:
Currently the project has struggled with a handful of bugs and
inconsistencies. Given the medical nature of the application it is
essential that we have confidence in our codebase acting in a reliable
way.

While critically important, high level end to end testing is difficult
and expensive to implement and maintain in the javascript ecosystem,
especially with cross platform technologies like React Native. We
currently have some test coverage, but we continue to need to add more.
Given the cost and difficulty of end to end test, is best practice to
rely on it for only the critical user flows and rely on unit tests and
static analysis for further test coverage as this balances the need to
maintain quality assurance and productivity in the development process.

Unit tests are easier to maintain, but currently our codebase does not
have every component or module covered. And again, these test, while
critically important, take time to implement and maintain. Additionally
unit test are better suited for ensuring higher level behavior of the
system, they are less suited for protecting against simple bugs such as
typos and implicit type coercion.

By introducing typescript we can quickly introduce some typesafety that
will help us maintain confidence that we are not shipping code that has
unexpected behavior. Typescript is incrementally adoptable and no
existing code needs to be updated immediately. To make a file be
captured by the typescript complier we need only change the extension
from .js to .ts.

This commit:
Adds the typescript configuration and adds a 'validate' script to
package json which will run the ts complier. Note that metro bundler
ships with typescript support and we are introducing any new build
infrastructure.

A single file was chosen to convert to .ts arbitrarily to validate the
tsc process works.

A future commit should include running this validation in CI.

Note that given the build processes provided by metro bundler, there is
no need to emit js code into a bundle as one would do in a web context.
Hence the inclusion of the --noEmit flag.

johnschoeman committed:
Include prettier in the validation script

Why:
Currently there are a handful of files that are not following the
project's prettierrc. This is likely due to individual developer
environments which are running prettier based off of global prettierrc
files rather than the projectns. While not of major concern for the
functionality of the application, this leads to a lot of git churn and
is ultimately a drag on productivity due to unnecessary overhead in the
code review process.

This commit:
Adds a prettier check to the validation script, such that it will fail
if the files are not formatted according to prettierrc.

johnschoeman committed:
Update husky to run validation script on push

Why:
Currently husky is running eslint and prettier on commit. This can be a
drag on commiting frequently discourages the practice. Additionally the
husky hooks do not run the jest test suite, meaning that it's likely
that prs can be opened which do not pass the jest test suite. Given that
this could be checked locally first, we are spending more CI cycles than
are necessary.

This commit:
Changes the husky git hoot to be pre-push rather than pre-commit and
adds yarn test to the process.

jeffstolz committed:
Update start and end screens, buttons, pickers, headers, backgrounds.
Remove illustrations and refactor existing code.

jeffstolz committed:
Update start and end screens, buttons, pickers, headers, backgrounds.
Remove illustrations and refactor existing code.

yonatanmk committed:
Remove location hashes from location upload JSON (#933)

#925 and #928 recently added location hashes to each location stored on the device.

When we retrieve these locations to export them to safe places we want to make sure we do not upload the hashes as well. This PR makes sure that we only upload a location's latitude longitude, and time to safe places.

JacobJaffe committed:
Remove In-App Location Permission Setting (#945)

* remove local location permisions

* lint

* revert header

Commit:
Fix incorrect strings for GPS product flavor. (#953)

johnschoeman commented on app/views/assessment/AssessmentStart.js:
```diff
@@ -1,18 +1,11 @@
  * @typedef { import("./Assessment").SurveyQuestion } SurveyQuestion
  * @typedef { import("./Assessment").SurveyOption } SurveyOption
  */
-import imageHealth from '../../assets/images/assessment/icon-health.png';
```
💯

johnschoeman reviewed (approved):
Looks like some nice clean up! well done! Only a quick glance, but it all looks good to me.

Commit:
Fix captcha background color

Commit:
Merge branch 'js-self-assessment-design' of github.com:Path-Check/covid-safe-paths into SAF-408/implement-self-reporting-screens

johnschoeman committed:
Request exposure notification authorization bt ios

Why:
We would like to use the exposure notification service provided by
apple in the application. In order to do so users must enable
authorization to the app. We would like to have this authorization
requested during the onboarding flow.

This commit:
Introduces a screen that will render in the BT app which will prompt the
user to enable Exposure Notifications. We introduced an
ExposureNotificationContext that contains the authorization state on the
js side and provides a way for components to request EN authorization by
calling into a Native Module which exposes a method for setting the
exposure notification to true in the iOS layer.

We also introduced the start of the Exposure Notification Service in the
BTE target and included the entitlements for Exposure Notifications.

Co-Authored-By: mattthousand matt.buckley@rocketinsights.com
Co-Authored-By: tparesi thomas.paresi@thoughtbot.com

johnschoeman committed:
Convert enable exposure notifications screen to ts

Why:
We would like to introduce typescript to the EnableExposureNotifications
screen which was added in a previous commit. In order to do so we needed
to update a fair amount of imported files to have types. These changes
where not included in that commit as they stand on their own and are
logically separated from the changes in that commit.

This commit:
Coverts imported the files which are imported to
EnableExposureNotifications to typescript. Some where changed
completely, others which included imports we added declarations files to
provide types such that this commit could remain smaller. Ideally in
future work we can remove the declaration files and incorporate the type
definitions directly in our application code.

Namit-Agrawal committed:
Fixed the difficulty in scrolling for android app  (#950)

* Fixed scrolling

* Fixed tests

* Made it compatible withbigger and small screens and removed from package.json

* Made UI cleaner and proportional on small and large screens

* Made a seperate component to display news item

* Cleaned up NewsItem and News

* Removed flexWrap from News

* Delete my_key

* Delete

* Fixed snapshot

* Multiple webviews now working

* Multiple webviews working

* Fixed tests

johnschoeman committed:
Update to prettier 2.0 and eslint with ts support

Why:
Currently our prettier and eslint setup do not support JSX within .tsx
files. We would like to update our prettier and eslint config to handle
this.

Prettier 2.0 was released in march and includes better typescript
support and eslint support.

This commit:
Updates prettier to 2.0 and adds all the required eslint plugins to
support tsx files.

Note that the prettier-plugin-import-sort is not updated for prettier
2.0 yet so it was removed.

johnschoeman committed:
Use prettier 2.0 default settings

Why:
The previous commit updated prettier to version 2.0.
https://prettier.io/blog/2020/03/21/2.0.0.html

The update included some changes to the default settings which largely
improve the development experience for typescript and we would like to
use these new defaults.

This commit:
Updates all of the files to use the new default settings.

JacobJaffe committed:
Fix GPS Staging Android APK builds (#956)

* update build name

* update list for all staging

Commit:
Merge branch 'js-self-assessment-design' of github.com:Path-Check/covid-safe-paths into SAF-408/implement-self-reporting-screens

yonatanmk committed:
Update Android Unit tests to use examples with timestamps in milliseconds (#954)

This PR updates a single android unit tests that had an example location timestamp in seconds. This should clarify any confusion about what unit timestamps should be in. Moving forward all location timestamps should be in milliseconds in both Safe Paths and Safe Places

Patrick-Erichsen committed:
General Code Cleanup (#924)

Note: This PR was originally to fix bugs on the Location Status indicator, but the UI for that element has changed and we are removing it altogether. I would still like to include some general cleanup that I did in the PR however.

Fixes onboarding bug where selecting Deny on location permissions blocks the user from continuing
Moves useFlags into a new services/hooks folder
Converts LocationData into a few functions instead of a class that needs to be instantiated
Creates an index.js for the components folder to allow for group imports. Cleans up all files that use a component to use a singular group import
Linked issues:
https://pathcheck.atlassian.net/browse/SAF-239
Screenshots:
N/A

How to test:
Walk through app and ensure no screens get a red box.


* Move HA data out of locationservice

Signed-off-by: Patrick Erichsen <patrick.a.erichsen@gmail.com>

* Add useLocTrackingStatus hook

Signed-off-by: Patrick Erichsen <patrick.a.erichsen@gmail.com>

* Add device specific toggling actions

Signed-off-by: Patrick Erichsen <patrick.a.erichsen@gmail.com>

* Cleanup onboarding5

Signed-off-by: Patrick Erichsen <patrick.a.erichsen@gmail.com>

* Refactor to switch/button status

Signed-off-by: Patrick Erichsen <patrick.a.erichsen@gmail.com>

* Rename Reason enum values

Signed-off-by: Patrick Erichsen <patrick.a.erichsen@gmail.com>

* Move EulaModal out of /components

Signed-off-by: Patrick Erichsen <patrick.a.erichsen@gmail.com>

* Fix Eula imports

Signed-off-by: Patrick Erichsen <patrick.a.erichsen@gmail.com>

* Fix broken unit tests

Signed-off-by: Patrick Erichsen <patrick.a.erichsen@gmail.com>

* Incorporate feedback

Signed-off-by: Patrick Erichsen <patrick.a.erichsen@gmail.com>

* Update snapshots

Signed-off-by: Patrick Erichsen <patrick.a.erichsen@gmail.com>

* Fix linting errors

Signed-off-by: Patrick Erichsen <patrick.a.erichsen@gmail.com>

* Revert to develop not master

Signed-off-by: Patrick Erichsen <patrick.a.erichsen@gmail.com>

* pull from develop

Signed-off-by: Patrick Erichsen <patrick.a.erichsen@gmail.com>

* better wording

Signed-off-by: Patrick Erichsen <patrick.a.erichsen@gmail.com>

* Undo TracingStrategyAsset changes

Signed-off-by: Patrick Erichsen <patrick.a.erichsen@gmail.com>

Commit:
Merge branch 'develop' of github.com:Path-Check/covid-safe-paths into SAF-408/implement-self-reporting-screens

Commit:
Merge branch 'develop' of github.com:Path-Check/covid-safe-paths into SAF-408/implement-self-reporting-screens

johnschoeman committed:
Use typescript plugin in eslint

Why:
A previous commit introduced typescript to the project. Currently the
eslint linter does not have support for typescript files.

This commit:
Adds a typescript-eslint plugin and adds a handful of rules for the
plugin.

johnschoeman committed:
Up timeout for unit-test-ios

Why:
Currently the github action, unit-test-ios, blocks merging if not green
and times out if the run takes longer than 15 minutes. It sometimes
takes longer than 15 minutes.

This commit:
Ups the timeout time. We should seriously consider the value of these
checks compared to the drag on productivity. It takes half a day to merge
an approved pr.

Commit:
iOS Key Server API Client (#960)

* Request exposure notification authorization bt ios

Why:
We would like to use the exposure notification service provided by
apple in the application. In order to do so users must enable
authorization to the app. We would like to have this authorization
requested during the onboarding flow.

This commit:
Introduces a screen that will render in the BT app which will prompt the
user to enable Exposure Notifications. We introduced an
ExposureNotificationContext that contains the authorization state on the
js side and provides a way for components to request EN authorization by
calling into a Native Module which exposes a method for setting the
exposure notification to true in the iOS layer.

We also introduced the start of the Exposure Notification Service in the
BTE target and included the entitlements for Exposure Notifications.

Co-Authored-By: mattthousand matt.buckley@rocketinsights.com
Co-Authored-By: tparesi thomas.paresi@thoughtbot.com

* Initial APIClient infrastructure for iOS.

* First pass at Exposure Notification iOS APIClient and core requests.

* Cleanup.

Co-authored-by: John Schoeman <johnschoeman1617@gmail.com>

JacobJaffe committed:
Global Store Introduction (#923)

* global store with persistence and onboarding as exmaple

* remove onboarding storage

* fix tests

* createStore -> createPersistedStore

* remove onboardinig 5

* Store -> store in mock

* redux toolkit

* cleanup

* undo lint changes

* lint

* prettier

* prettier

* prettier

Commit:
Merge branch 'develop' of github.com:Path-Check/covid-safe-paths into SAF-408/implement-self-reporting-screens

Commit:
Fix removed assessment screen from navigation stack

JacobJaffe committed:
Connect Export To Staging Server & Redux Typescript (#964)

Connects export to Staging Server! Also adds API layer, and converts redux layer to typescript.

Adds connection to the new trusted-authorities repo for HA selection during export.
Next steps are to use that everywhere.

Currently the HA yaml is hard coded (fine because this is all feature flagged still), this should ideally be a env variable to point to different yamls depending on lane.

This also makes everything under the /store folder typescript, so that our data layer is less brittle.

Adds linking to privacy policy. Some design divergence (whole line is a link instead of just 'here') here to make this accessible across localization. We can revisit this and introduce the i18next trans component as a follow up if we want to specify a word across locals.

JacobJaffe committed:
onboarding permisions hotfix for bte (#970)

Commit:
Merge branch 'develop' of github.com:Path-Check/covid-safe-paths into SAF-408/implement-self-reporting-screens

Commit:
Update yarn.lock

Commit:
Prettier fix

Commit:
Rename svg files to ts

Commit:
Fix lint errors

Commit:
Fix prettier errors

Commit:
Fix tests
